src/encoded/types/patient.py

```diff
@@ -460,11 +438,43 @@ def surgery_summary(self, request, surgery=None):
                 surgery_summary = "No Treatment Received"
             return surgery_summary
 
+   
+
     @calculated_property(schema={
-        "title": "Diagnosis Date",
-        "type": "string",
+        "title": "Diagnosis",
+        "description": "Infomation related to diagnosis",
+        "type": "object",
+        "additionalProperties": False,
+        "properties":{
+            "diagnosis_date": {
+                "title": "Diagnosis Date",
+                "description": "Date of Diagnosis",
+                "type": "string",
+            },
+            "age": {
+                "title": "Diagnosis age",
+                "description": "The age of diagnosis.",
+                "type": "string",
+                "pattern": "^((\\d+(\\.[1-9])?(\\-\\d+(\\.[1-9])?)?)|(unknown)|([1-8]?\\d)|(90 or above))$"
```

Comment:
What does this pattern mean? ((\\d+(\\.[1-9])?(\\-\\d+(\\.[1-9])?)?)|([1-8]?\\d)

Reply:
Simplify the regular expression to be from and only have units has year. 
0-89
unknown
90 or above

Reply:
Done